### DISCUSSION TITLE
REST API for version of code

### DISCUSSION THREAD

**Discussion by sagoresarker** [community/community#22361] [New to GitHub]
<p>I’m working on a project where I need to extract the version of code using GitHub rest API. But I can’t find it.</p>
<p>IntelliJ and many of the existing IDE use the API for showing the differences. I explored many of these to find the API. But didn’t find it yet.</p>
<p>What is the API for extracting the different versions of code?</p>

**Comment by jfullstackdev** [comment]
<p>are you referring to the releases of code on GitHub?</p>
<p>check this out,<br>
<a href="https://docs.github.com/en/rest/reference/repos#releases" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/repos#releases</a></p>

**Comment by sagoresarker** [comment]
<p>Thanks for your information. I really want to fetch the user’s commit history along with the different versions of code through GitHub API.</p>
<p>As an overview of my university project,  I have to build a website that is capable of getting input from users GitHub repo link, fetching data from GitHub, and showing the latest 9-10 changes in codes he uploaded on GitHub, in my website. In short, it’ll be a web based code comparing tool.</p>

**Comment by jfullstackdev** [comment]
<p>they are scattered at the API, but the one you will be using heavily is <code>Commits</code></p>
<p><a href="https://docs.github.com/en/rest/reference/commits" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/commits</a></p>
<p>your idea is just like viewing the entire commits and all it’s changes</p>

**Comment by sagoresarker** [comment]
<p>Thanks a lot. By using this API, will I get all the versions of code pointed by the input URL?</p>

**Comment by jfullstackdev** [comment, marked as answer]
<p>yes (unless the API limits that),</p>
<p>for example</p>
<p>you can get all the commits,</p>
<p><a href="https://docs.github.com/en/rest/reference/commits#list-commits" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/commits#list-commits</a></p>
<p>and get a commit, if you want to dig deeper,</p>
<p><a href="https://docs.github.com/en/rest/reference/commits#get-a-commit" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/commits#get-a-commit</a></p>
